src/OrchardCore.Modules/OrchardCore.Autoroute/Views/AutoroutePartSettings.Edit.cshtml

```diff
@@ -1,9 +1,18 @@
 @model OrchardCore.Autoroute.ViewModels.AutoroutePartSettingsViewModel
 
+<script asp-name="codemirror" depends-on="admin" at="Foot"></script>
+<script asp-name="codemirror-mode-javascript" at="Foot"></script>
+<script asp-name="codemirror-addon-display-autorefresh" at="Foot"></script>
+<script asp-name="codemirror-addon-mode-simple" at="Foot"></script>
+<script asp-name="codemirror-addon-mode-multiplex" at="Foot"></script>
+<script asp-name="codemirror-mode-xml" at="Foot"></script>
+
+<script asp-src="~/OrchardCore.Liquid/codemirror/liquid.js" at="Foot"></script>
+
 <div class="form-group row">
     <div class="col-lg">
         <label asp-for="Pattern">@T["Pattern"]</label>
-        <input asp-for="Pattern" class="form-control" />
+        <textarea asp-for="Pattern" rows="5" class="form-control"></textarea>
```

Comment:
We need a mode that has an auto-growth such that it will text as much space as necessary. We had that in O1

Reply:
I removed rows="5" and it displays the proper amount of lines but then you mean for width?

Reply:
I meant for lines

Reply:
![screencapture-localhost-44300-Admin-ContentTypes-Acme-ContentParts-AutoroutePart-Edit-2019-06-03-18_34_29](https://user-images.githubusercontent.com/3228637/58838982-407c0d00-862e-11e9-95ec-2fc486f1f83a.png)

Reply:
With or without rows="something" I get the same result.

Reply:
We need to remove `\r\n` from the generated path.